src/tests/omni_lock_util.rs

```diff
@@ -0,0 +1,181 @@
+        }
+    };
+
+    // it not needed to set "type script" when is_type is false
+    let capacity = bin.len() as u64;
+    let output = CellOutput::new_builder()
+        .capacity(capacity.pack())
+        .lock(always_success)
+        .type_(Some(type_script_in_code.clone()).pack())
+        .build();
+    let out_point = random_out_point();
+    let cell_dep = CellDep::new_builder()
+        .out_point(out_point.clone())
+        .dep_type(DepType::Code.into())
+        .build();
+    ctx.add_cell_dep(cell_dep, output, bin.clone(), None);
+    rce_cells.push(out_point);
+
+    let code_hash = if is_type {
+        ckb_hash::blake2b_256(type_script_in_code.as_slice())
+    } else {
+        ckb_hash::blake2b_256(bin)
+    };
+    let hash_type = if is_type {
+        ScriptHashType::Type
+    } else {
+        ScriptHashType::Data
+    };
+
+    Script::new_builder()
+        .args(args.pack())
+        .code_hash(code_hash.pack())
+        .hash_type(hash_type.into())
+        .build()
+}
+
+// first generate N RCE cells with each contained one RCRule
+// then collect all these RCE cell hash and create the final RCE cell.
+pub fn generate_rce_cell(
+    ctx: &mut Context,
+    rc_rules: Vec<Bytes>,
+    rce_cells: &mut Vec<OutPoint>,
+) -> Byte32 {
+    let mut rng = thread_rng();
+    let mut cell_vec_builder = RCCellVecBuilder::default();
+
+    for rc_rule in rc_rules {
+        let mut random_args: [u8; 32] = Default::default();
+        rng.fill(&mut random_args[..]);
+        let rce_script = build_script(
+            ctx,
+            true,
+            false,
+            &rc_rule,
+            Bytes::copy_from_slice(random_args.as_ref()),
+            rce_cells,
+        );
+
+        let hash = rce_script.code_hash();
+
+        cell_vec_builder =
+            cell_vec_builder.push(Byte32::from_slice(hash.as_slice()).expect("Byte32::from_slice"));
+    }
+
+    let cell_vec = cell_vec_builder.build();
+
+    let rce_cell_content = RCDataBuilder::default()
+        .set(RCDataUnion::RCCellVec(cell_vec))
+        .build();
+
+    let mut random_args: [u8; 32] = Default::default();
+    rng.fill(&mut random_args[..]);
+    let bin = rce_cell_content.as_slice();
+    let rce_script = build_script(
+        ctx,
+        true,
+        false,
+        &Bytes::copy_from_slice(bin),
+        Bytes::copy_from_slice(random_args.as_ref()),
+        rce_cells,
+    );
+    rce_script.code_hash()
+}
+
+pub fn add_rce_cells(
+    tx: ckb_types::core::TransactionView,
+    rce_cells: &[OutPoint],
+) -> ckb_types::core::TransactionView {
+    if !rce_cells.is_empty() {
+        let mut builder = tx.as_advanced_builder();
+        for cell in rce_cells {
+            builder = builder.cell_dep(
+                CellDep::new_builder()
+                    .out_point(cell.clone())
+                    .dep_type(DepType::Code.into())
+                    .build(),
+            );
+        }
+        return builder.build();
+    }
+    tx
```

Comment:
Code in test file better as simple as possible:
```suggestion
    let mut builder = tx.as_advanced_builder();
    for cell in rce_cells {
        builder = builder.cell_dep(
            CellDep::new_builder()
                .out_point(cell.clone())
                .dep_type(DepType::Code.into())
                .build(),
        );
    }
    builder.build()
```